src/drivers/intel/ssp/ssp.c

```diff
@@ -690,6 +685,35 @@ static int ssp_pre_start(struct dai *dai)
 	return ret;
 }
 
+/*
+ * Portion of the SSP configuration should be applied just before the
+ * SSP dai is activated, for either power saving or params runtime
+ * configurable flexibility.
+ */
+static int ssp_pre_start(struct dai *dai)
+{
+	struct ssp_pdata *ssp = dai_get_drvdata(dai);
+	int ret = 0;
+
+	dai_info(dai, "ssp_pre_start()");
+
+	/* SSP active means bclk already configured. */
+	if (ssp->state[SOF_IPC_STREAM_PLAYBACK] == COMP_STATE_ACTIVE ||
+	    ssp->state[SOF_IPC_STREAM_CAPTURE] == COMP_STATE_ACTIVE)
```

Comment:
I don't think that test is valid for the early start. the ACTVE state is set in the trigger case.

Reply:
> I don't think that test is valid for the early start. the ACTVE state is set in the trigger case.

`ssp_pre_start()` is only called by `ssp_start()` now, so it is not used for the early start.